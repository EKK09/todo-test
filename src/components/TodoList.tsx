import React, { useEffect } from 'react';
import useTodoListStore from 'src/store/todoList';
import styled from 'styled-components';
import TodoListItem from './TodoListItem';

const LoadingWrapper = styled.div`
  font-weight: bolder;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-wrap: nowrap;
`;

const TodoList = (): React.ReactElement => {
  const { todoList, isFetching, fetchTodoList } = useTodoListStore(
    (state) => ({
      todoList: state.todoList,
      isFetching: state.isFetching,
      fetchTodoList: state.fetchTodoList,
    }),
  );
  const todoListItems = todoList.map((todo) => <TodoListItem key={todo.id} todo={todo} />);

  useEffect(() => {
    fetchTodoList();
  }, []);

  if (isFetching) {
    return <LoadingWrapper>Loading ....</LoadingWrapper>;
  }

  return (
    <div>
      {todoListItems}
    </div>
  );
};

export default TodoList;
