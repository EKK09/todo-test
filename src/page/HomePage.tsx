import React from 'react';
import TodoList from 'src/components/TodoList';

const HomePage = (): React.ReactElement => (
  <div>
    <TodoList />
  </div>
);

export default HomePage;
