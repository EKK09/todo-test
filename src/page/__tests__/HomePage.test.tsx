import React from 'react';
import { render } from '@testing-library/react';
import { create } from 'react-test-renderer';
import TodoList from 'src/components/TodoList';
import HomePage from '../HomePage';

describe('HomePage component', () => {
  it('Render HomePage', async () => {
    const wrapper = render(<HomePage />);

    expect(wrapper.container).toMatchSnapshot();
  });

  it('should render Todolist component', async () => {
    const wrapper = create(<HomePage />);
    const todoListComponents = wrapper.root.findAllByType(TodoList);

    expect(todoListComponents.length).toBe(1);
  });
});
