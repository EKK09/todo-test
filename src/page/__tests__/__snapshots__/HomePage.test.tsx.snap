// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component Render HomePage 1`] = `
<div>
  <div>
    <div />
  </div>
</div>
`;
